kotest-assertions/kotest-assertions-json/src/jvmTest/kotlin/com/sksamuel/kotest/tests/json/ContainJsonKeyValueTest.kt

```diff
@@ -0,0 +1,89 @@
+package com.sksamuel.kotest.tests.json
+
+import io.kotest.assertions.asClue
+import io.kotest.assertions.json.shouldContainJsonKeyValue
+import io.kotest.assertions.json.shouldNotContainJsonKeyValue
+import io.kotest.assertions.shouldFail
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+import org.intellij.lang.annotations.Language
+
+class ContainJsonKeyValueTest : StringSpec({
+   @Language("JSON")
+   val json = """
+      {
+          "store": {
+              "book": [
+                  {
+                    "category": "reference",
+                    "author": "Nigel Rees",
+                    "title": "Sayings of the Century",
+                    "price": 8.95
+                 },
+                 {
+                    "category": "fiction",
+                    "author": "Evelyn Waugh",
+                    "title": "Sword of Honour",
+                    "price": 12.99
+                 }
+              ],
+              "bicycle": {
+                 "color": "red",
+                 "price": 19.95,
+                 "code": 1
+              }
+         }
+      }
+   """.trimIndent()
+
+   "Negated assertions" {
+      "".shouldNotContainJsonKeyValue("$.store.bicycle.color", "red")
+      "{}".shouldNotContainJsonKeyValue("$.store.bicycle.color", "red")
+      """{ "foo": "bar" }""".shouldNotContainJsonKeyValue("foo", "baz")
+      shouldFail {
+         """{ "foo": "bar" }""".shouldNotContainJsonKeyValue("foo", "bar")
+      }.message shouldBe """{ "foo": "bar" } should not contain the element foo = bar"""
+   }
+
+   "Failure message states if key is missing, when it's missing" {
+      shouldFail {
+         json.shouldContainJsonKeyValue("$.bicycle.engine", "V2")
+      }.message shouldBe """
+         Expected given to contain json key <'$.bicycle.engine'> but key was not found.
```

Comment:
Would it make sense to print partial matches, such as:
```
Expected given to contain json key <'$.store.book.weight'> but key was not found. 
Found the following sub-key: <'$.store.book'>.
```

Reply:
I suppose it would, but I don't believe we get that information out of the JsonPath library. How do you intend that we should calculate that subkey?

Reply:
I think perhaps we can merge this small improvement as-is, and we could make further improvements in a separate PR :) We can continue the discussion here even after PR is merged

Reply:
> I think perhaps we can merge this small improvement as-is, and we could make further improvements in a separate PR :) We can continue the discussion here even after PR is merged

sure - https://github.com/kotest/kotest/pull/4013